src/components/MoneyRequestConfirmationList.js

```diff
@@ -88,244 +88,219 @@ const defaultProps = {
      * @returns {Array}
      */
-    getSplitOrRequestOptions() {
-        const text = this.props.translate(this.props.hasMultipleParticipants ? 'iou.splitAmount' : 'iou.requestAmount', {
-            amount: CurrencyUtils.convertToDisplayString(this.props.iouAmount, this.props.iou.selectedCurrencyCode),
+    const getParticipantsWithAmount = useCallback(
+        (participantsList) => {
+            const iouAmount = IOUUtils.calculateAmount(participantsList.length, props.iouAmount);
+            return OptionsListUtils.getIOUConfirmationOptionsFromParticipants(participantsList, CurrencyUtils.convertToDisplayString(iouAmount, props.iou.selectedCurrencyCode));
+        },
+        [props.iouAmount, props.iou.selectedCurrencyCode],
+    );
+
+    const formattedParticipants = _.map(getParticipantsWithAmount(props.participants), (participant) => ({
+        ...participant,
+        selected: true,
+    }));
+
+    const [participants, setParticipants] = useState(formattedParticipants);
+    const [didConfirm, setDidConfirm] = useState(false);
+
+    const splitOrRequestOptions = useMemo(() => {
+        const text = props.translate(props.hasMultipleParticipants ? 'iou.splitAmount' : 'iou.requestAmount', {
+            amount: CurrencyUtils.convertToDisplayString(props.iouAmount, props.iou.selectedCurrencyCode),
         });
         return [
             {
                 text: text[0].toUpperCase() + text.slice(1),
-                value: this.props.hasMultipleParticipants ? CONST.IOU.MONEY_REQUEST_TYPE.SPLIT : CONST.IOU.MONEY_REQUEST_TYPE.REQUEST,
+                value: props.hasMultipleParticipants ? CONST.IOU.MONEY_REQUEST_TYPE.SPLIT : CONST.IOU.MONEY_REQUEST_TYPE.REQUEST,
             },
         ];
-    }
+    }, [props.hasMultipleParticipants, props.iouAmount, props.iou.selectedCurrencyCode, props.translate]);
 
-    /**
-     * Get selected participants
-     * @returns {Array}
-     */
-    getSelectedParticipants() {
-        return _.filter(this.state.participants, (participant) => participant.selected);
-    }
-
-    /**
-     * Get unselected participants
-     * @returns {Array}
-     */
-    getUnselectedParticipants() {
-        return _.filter(this.state.participants, (participant) => !participant.selected);
-    }
-
-    /**
-     * Returns the participants with amount
-     * @param {Array} participants
-     * @returns {Array}
-     */
-    getParticipantsWithAmount(participants) {
-        const iouAmount = IOUUtils.calculateAmount(participants.length, this.props.iouAmount);
-        return OptionsListUtils.getIOUConfirmationOptionsFromParticipants(participants, CurrencyUtils.convertToDisplayString(iouAmount, this.props.iou.selectedCurrencyCode));
-    }
+    const selectedParticipants = useMemo(() => _.filter(participants, (participant) => participant.selected), [participants]);
+    const unselectedParticipants = useMemo(() => _.filter(participants, (participant) => !participant.selected), [participants]);
```

Comment:
`unselectedParticipants` seems that is being used only for `optionSelectorSections` which is a use memo, so if `participants` changes, `unselectedParticipants` will change, and will re-run `optionSelectorSections`. I would move the filter for `unselectedParticipants` inside of `optionSelectorSections`

Reply:
I think it makes sense to move `unselectedParticipants` inside `optionSelectorSections ` since it's only used there. However, `optionSelectorSections` will then also have `participants` as a dependency, which will re-run it anyways.

Reply:
Yeah, I mean the logic to move inside, because `optionSelectorSections` will run when `participants` change, so you can do the filtering inside of `optionSelectorSections` before using it